### PR TITLE
Add to support `lein cloverage` (reformed)

### PR DESCRIPTION
**Pull request by ayamada**
Please apply #44 before this.

Add `lein cloverage`.
This patch includes some fixes for `lein cloverage`.

- Importing `deftype`d classes should need `require` in combination.
- `extend-type` (and `extend`) do `alter-var-root` internally. So `extend-type` should be just below `deftype`.
- Don't run `:slow` and `:heavy` tests in `lein cloverage`.
    - Related link : https://github.com/cloverage/cloverage/issues/54

Diff of #44 and this branch : https://github.com/ayamada/cljam/compare/feature/migrate-clojure-test-for-pr-3...feature/support-lein-cloverage-for-pr-2

### CODE REVIEW

File: src/cljam/bam/reader.clj [hunk 1]
```diff
@@ -17,6 +18,40 @@
   (close [this]
     (.close ^Closeable (.reader this))))
 
+(declare read-alignments-sequentially*
+         read-alignments*
+         read-blocks-sequentially*
+         read-blocks*)
+
+(extend-type BAMReader
```
Comment (alumi) [inline src/cljam/bam/reader.clj]: `extend-type` can be merged with `deftype` and I think that'll be more clear.

File: test/cljam/t_common.clj [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
             [cljam.bam :as bam]
             [cljam.io :as io]
             [cavia.core :as cavia :refer [defprofile with-profile]]))
 
+(defn- _in-cloverage? []
+  (let [in? (try
+              (eval '(var cloverage.coverage/*covered*))
+              true
+              (catch Throwable _ nil))]
+    ;; (prn :debug 'in-cloverage? in?)
```
Comment (totakke) [inline test/cljam/t_common.clj]: Is this line a debug print? If so, please remove the line and make this function simple.

File: test/cljam/t_common.clj [hunk 3]
```diff
@@ -1,11 +1,35 @@
...
+    in?))
+
+(def ^:private in-cloverage? (memoize _in-cloverage?))
+
+(defn- expand-deftest [sym args bodies]
```
Comment (totakke) [inline test/cljam/t_common.clj]: Please rename `bodies` `body`. Also `deftest-slow`, `deftest-heavy`, and `deftest-slow-heavy`.

cf. [clojure-style-guide#idiomatic-names](https://github.com/bbatsov/clojure-style-guide#idiomatic-names)

Comment (ayamada) [inline test/cljam/t_common.clj]: Thanks to your review.
I fixed these defects by https://github.com/chrovis/cljam/pull/45/commits/26e57312a55d17a8cffba6e4e1da39c2ae9fa9cc .

File: src/cljam/bam/reader.clj (hunk 1 above)
Comment (ayamada) [inline src/cljam/bam/reader.clj]: Thanks to your review.
I fixed this by https://github.com/chrovis/cljam/pull/45/commits/f444f0249fa2e8046b8a48030943157e53f4ddc3 .

Comment (alumi) [inline src/cljam/bam/reader.clj]: Nice work! Thanks!

**Comment by totakke** [comment]
Perfect. Thanks.

**Pull request merged**